src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -1212,45 +1212,165 @@ bool ViewProviderSketch::mouseButtonPressed(int Button, bool pressed, const SbVe
                     return true;
                 case STATUS_NONE: {
                     // A right click shouldn't change the Edit Mode
-                    if (preselection.isPreselectPointValid()) {
-                        return true;
+                    bool noSelection = true;
+                    bool selectPoint = false;
+                    bool selectLine = false;
+                    bool selectConstraint = false;
+                    bool selectMultiple = false;
+
+                    Gui::MenuItem geom;
+                    geom.setCommand("Sketcher context");
+                    // Gets a selection vector
+                    std::vector<Gui::SelectionObject> selection = Gui::Selection().getSelectionEx();
+                    // if something is selected
+                    if (selection.size() > 0) {
+                        const std::vector<std::string> SubNames = selection[0].getSubNames();
+
+                    if (SubNames.size() == 1) {
```

Comment:
From here you are ignoring the indentation from the previous if. This makes the code hard to understand as one can't visually see where the if starts and ends. Besides it is misleading to the reader. Please fix the indentation.